source/common/json/json_loader.cc

```diff
@@ -1,178 +1,194 @@
-  json_t* string = json_object_get(json_, name.c_str());
-  if (!string || !json_is_string(string)) {
-    throw Exception(fmt::format("key '{}' missing or not a string in '{}'", name, name_));
+    std::vector<std::string> string_array;
+    for (rapidjson::SizeType i = 0; i < document_[name.c_str()].Size(); i++) {
+      if (!document_[name.c_str()][i].IsString()) {
+        throw Exception(fmt::format("array '{}' does not contain all strings", name));
+      }
+      string_array.push_back(document_[name.c_str()][i].GetString());
+    }
+    return string_array;
   }
 
-  return json_string_value(string);
-}
+  double getDouble(const std::string& name) const override {
+    if (!document_[name.c_str()].IsDouble()) {
+      throw Exception(fmt::format("key '{}' missing or not a double in '{}'", name, name_));
+    }
 
-std::string Object::getString(const std::string& name, const std::string& default_value) const {
-  if (!json_object_get(json_, name.c_str())) {
-    return default_value;
-  } else {
-    return getString(name);
+    return document_[name.c_str()].GetDouble();
   }
-}
 
-std::vector<std::string> Object::getStringArray(const std::string& name) const {
-  std::vector<std::string> string_array;
-  for (Object& object : getObjectArray(name)) {
-    if (!json_is_string(object.json_)) {
-      throw Exception(fmt::format("array '{}' does not contain all strings", name));
+  double getDouble(const std::string& name, double default_value) const override {
+    if (!document_.HasMember(name.c_str())) {
+      return default_value;
     } else {
-      string_array.push_back(json_string_value(object.json_));
+      return getDouble(name);
     }
   }
 
-  return string_array;
-}
-
-double Object::getDouble(const std::string& name) const {
-  json_t* real = json_object_get(json_, name.c_str());
-  if (!real || !json_is_real(real)) {
-    throw Exception(fmt::format("key '{}' missing or not a double in '{}'", name, name_));
+  void iterate(const AbstractObjectCallback& callback) override {
+    for (rapidjson::Value::ConstMemberIterator itr = document_.MemberBegin();
+         itr != document_.MemberEnd(); ++itr) {
+      std::string object_key(itr->name.GetString());
+      AbstractObjectImpl object_value(itr->value, object_key);
+      bool need_continue = callback(object_key, object_value);
+      if (!need_continue) {
+        break;
+      }
+    }
+  }
+  bool hasObject(const std::string& name) const override {
+    return document_.HasMember(name.c_str());
   }
 
-  return json_real_value(real);
-}
-
-double Object::getDouble(const std::string& name, double default_value) const {
-  if (!json_object_get(json_, name.c_str())) {
-    return default_value;
-  } else {
-    return getDouble(name);
+  std::vector<AbstractObjectPtr> getMembers() const override {
+    std::vector<AbstractObjectPtr> return_vector;
+    for (rapidjson::Value::ConstMemberIterator itr = document_.MemberBegin();
```

Comment:
if this is how getMembers() is implemented I don't understand why we need a new getMembers() method at all. Just use this to implement iterate.